Minor updates to improve consistency in README.md (#16390)
hashicorp/vault: README.md

The key features of Vault are:

> * **Secure Secret Storage**: Vault can store arbitrary key/value secrets. Vault encrypts these secrets prior to writing them to persistent

Reviewer: ```suggestion
* **Secure Secret Storage**: Vault can store arbitrary key/value pairs. Vault encrypts data before writing it to persistent
```

Suggested rewrite to reduce the number of times we use "secrets" in the same sentence

  developers to store encrypted data in a location such as a SQL database without
  having to design their own encryption methods.

> * **Leasing and Renewal**: Vault associates a _lease_ with each secret.

Reviewer: ```suggestion
* **Leasing and Renewal**: Vault associates a **lease** with each secret.
```

Style correction: avoid italics in favor of bolding

>   At the end of the lease, Vault will automatically revoke that

Reviewer: ```suggestion
  At the end of the lease, Vault automatically revokes the
```
Style correction: write in active voice
